electron/launcher.ts

```diff
@@ -96,300 +62,322 @@ async function launch(
-      ? 'DRI_PRIME=1 __NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia'
-      : '',
-    proton: isProton
-      ? `STEAM_COMPAT_CLIENT_INSTALL_PATH=${steamCompatFolder} STEAM_COMPAT_DATA_PATH='${winePrefix
-          .replaceAll("'", '')
-          .replace('~', userHome)}'`
-      : ''
+  // Log warning about Proton
+  if (gameSettings.wineVersion.type == 'proton') {
+    logWarning(
+      'You are using Proton, this can lead to some bugs. Please do not open issues with bugs related to games',
+      LogPrefix.Backend
+    )
   }
 
-  envVars = Object.values(options)
-    .filter((n) => n)
-    .join(' ')
-  if (isProton) {
-    logWarning(
-      [
-        `You are using Proton, this can lead to some bugs,
-          please do not open issues with bugs related with games`,
-        wineVersion.name
-      ],
+  const { updated: winePrefixUpdated } = await verifyWinePrefix(game)
+  if (winePrefixUpdated) {
+    logInfo(
+      ['Created/Updated Wineprefix at', gameSettings.winePrefix],
       LogPrefix.Backend
     )
+    setup(game.appName)
+  }
+
+  // If DXVK/VKD3D installation is enabled, install it
+  if (gameSettings.wineVersion.type == 'wine') {
+    if (gameSettings.autoInstallDxvk) {
+      await DXVK.installRemove(
+        gameSettings.winePrefix,
+        gameSettings.wineVersion.bin,
+        'dxvk',
+        'backup'
+      )
+    }
+    if (gameSettings.autoInstallVkd3d) {
+      await DXVK.installRemove(
+        gameSettings.winePrefix,
+        gameSettings.wineVersion.bin,
+        'vkd3d',
+        'backup'
+      )
+    }
   }
 
-  await createNewPrefix(isProton, fixedWinePrefix, winePath, appName)
+  const envVars = setupWineEnvVars(gameSettings)
 
-  // Install DXVK for non Proton/CrossOver Prefixes
-  if (!isProton && !isCrossover && autoInstallDxvk) {
-    await DXVK.installRemove(winePrefix, wineVersion.bin, 'dxvk', 'backup')
+  return { success: true, envVars: envVars }
+}
+
+/**
+ * Maps general settings to environment variables
+ * @param gameSettings The GameSettings to get the environment variables for
+ * @returns A big string of environment variables, structured key=value
+ */
+function setupEnvVars(gameSettings: GameSettings) {
+  const ret: Record<string, string> = {}
+  if (gameSettings.nvidiaPrime) {
+    ret.DRI_PRIME = '1'
+    ret.__NV_PRIME_RENDER_OFFLOAD = '1'
+    ret.__GLX_VENDOR_LIBRARY_NAME = 'nvidia'
+  }
+  if (gameSettings.audioFix) {
+    ret.PULSE_LATENCY_MSEC = '60'
   }
 
-  if (wineVersion.name !== 'Wine Default') {
-    const { bin } = wineVersion
-    wineCommand = isProton
-      ? `--no-wine --wrapper "${bin} run"`
-      : `--wine ${bin}`
+  return ret
+}
+
+/**
+ * Maps Wine-related settings to environment variables
+ * @param gameSettings The GameSettings to get the environment variables for
+ * @returns A big string of environment variables, structured key=value
+ */
+function setupWineEnvVars(gameSettings: GameSettings) {
+  let ret: Record<string, string> = {}
```

Comment:
what `ret` means? 👀

Reply:
Short for `return`, so it's the object that's returned by the function  
Very creative, I know